boot/bootstate16.go

```diff
@@ -150,3 +150,38 @@ func (s *bootState16) markSuccessful(update bootStateUpdate) (bootStateUpdate, e
 
 	return u16, nil
 }
+
+func (s16 *bootState16) setNext(nextBoot string) (rebootRequired bool, u bootStateUpdate, err error) {
+	nextBootVar := fmt.Sprintf("snap_try_%s", s16.varSuffix)
+	goodBootVar := fmt.Sprintf("snap_%s", s16.varSuffix)
+
+	u16, err := newBootStateUpdate16(nil, "snap_mode", goodBootVar)
+	if err != nil {
+		return false, nil, err
+	}
+
+	env := u16.env
+	toCommit := u16.toCommit
+
+	snapMode := "try"
+	rebootRequired = true
+	if env[goodBootVar] == nextBoot {
+		// If we were in anything but default ("") mode before
+		// and now switch to the good core/kernel again, make
```

Comment:
the `now` makes this statement a little hard to parse

```suggestion
		// and switched to the good core/kernel again, make
```